sklearn/ensemble/_iforest.py

```diff
@@ -195,6 +197,28 @@ class IsolationForest(OutlierMixin, BaseBagging):
     array([ 1,  1, -1])
     """
 
+    _parameter_constraints = {
+        "n_estimators": [Interval(Integral, 1, None, closed="left")],
+        "max_samples": [
+            StrOptions({"auto"}),
+            Interval(Integral, 1, None, closed="left"),
+            Interval(Real, 0, 1, closed="right"),
+        ],
+        "contamination": [
+            StrOptions({"auto"}),
+            Interval(Real, 0, 0.5, closed="right"),
+        ],
+        "max_features": [
+            Integral,
+            Interval(Real, 0, 1, closed="right"),
+        ],
+        "bootstrap": [bool],
+        "n_jobs": [Integral, None],
+        "random_state": ["random_state"],
+        "verbose": [Integral],
```

Comment:
We need to have https://github.com/scikit-learn/scikit-learn/pull/23723 integrated first but it should become

```suggestion
        "verbose": [Integral],
```